api/test/unit/validation/attributeValidator.test.js

```diff
@@ -13,31 +13,56 @@ describe('Validating Policy JSON properties',function(){
     beforeEach(function(){
       fakePolicy = JSON.parse(fs.readFileSync(__dirname+'/../fakePolicy.json', 'utf8'));
     });
-    
+
     it('Should validate the policy JSON successfully',function(){
       attributeValidator.validatePolicy(fakePolicy,function(result){
         expect(result).to.be.empty;
       });
     });
-    it('Should fail to validate the policy as instance_min_count is greater than instance_max_count with schedule defined',function(){
-      fakePolicy.instance_min_count = 10;
-      fakePolicy.instance_max_count = 5;
-      attributeValidator.validatePolicy(fakePolicy,function(result){
-        expect(result[0]).to.have.property('stack').and.equal('instance_min_count 10 is higher or equal to instance_max_count 5 in policy_json');
-        expect(result[0]).to.have.property('message').and.equal('instance_min_count and instance_max_count values are not compatible');
-        expect(result[0]).to.have.property('property').and.equal('instance_min_count');
+
+    context('Validating instance_min/max_count properties',function(){
+      beforeEach(function(){
+        fakePolicy.instance_min_count = 10;
+        fakePolicy.instance_max_count = 5;
       });
+
+      it('Should fail to validate the policy as instance_min_count is greater than instance_max_count with schedule defined',function(){
+          attributeValidator.validatePolicy(fakePolicy,function(result){
+            expect(result[0]).to.have.property('stack').and.equal('instance_min_count 10 is higher or equal to instance_max_count 5 in policy_json');
+            expect(result[0]).to.have.property('message').and.equal('instance_min_count and instance_max_count values are not compatible');
+            expect(result[0]).to.have.property('property').and.equal('instance_min_count');
+          });
+        });
+
+        it('Should fail to validate the policy as instance_min_count is greater than instance_max_count without schedule defined',function(){
+          fakePolicy.schedules = null
```

Comment:
do we need to test "undefined"?

Reply:
it is a regression case. 
we did have validate error when schedules is undefined previously.
These case didn't change, just reformatted with "context"

Reply:
change to "delete fakePolicy.schedules" to test the situation of "undefined"..

Also, add more testcase to validate the policy when properties are set to null.